code/02_agg_climate_data.R

```diff
@@ -0,0 +1,240 @@
+## Sara Orofino
+## February 22, 2022
+## Aggregate Climate Data: Pipeline Step 02
+
+agg_climate_data <- function(year, data_source, climate_var, daily_agg, trans = 'polynomial', trans_specs, weights) {
+  
+  ## Setup 
+  ## -----------------------------------------------
+  require(pacman)
+  pacman::p_load(ncdf4, data.table, rgdal, raster, tidyverse, here, crayon)
+
+  # Function to convert raster to data.table from https://gist.github.com/etiennebr/9515738
+  as.data.table.raster <- function(x, row.names = NULL, optional = FALSE, xy=FALSE, inmem = canProcessInMemory(x, 2), ...) {
+    stopifnot(require("data.table"))
+    if(inmem) {
+      v <- as.data.table(as.data.frame(x, row.names=row.names, optional=optional, xy=xy, ...))
+    } else {
+      tr <- blockSize(x, n=2)
+      l <- lapply(1:tr$n, function(i) 
+        as.data.table(as.data.frame(getValues(x, 
+                                              row=tr$row[i], 
+                                              nrows=tr$nrows[i]), 
+                                    row.names=row.names, optional=optional, xy=xy, ...)))
+      v <- rbindlist(l)
+    }
+    coln <- names(x)
+    if(xy) coln <- c("x", "y", coln)
+    setnames(v, coln)
+    v
+  }
+  setMethod('as.data.table', signature(x='Raster'), as.data.table.raster)
+  
+  ## Load saved weights 
+  ## -----------------------------------------------
+  
+  # Normalize the data source input - remove spaces & lower case
+  data_source_norm <- gsub(" ", "", data_source) %>% tolower(.)
+  
+  # File name to call the weights
+  if(weights){
+    weights_file <- paste0(paste(input_polygons_name, data_source_norm, "area", weights_type, "weights", sep="_"), ".csv")
+  } else{
+    weights_file <- paste0(paste(input_polygons_name, data_source_norm, "area", "weights", sep="_"), ".csv")
+  }
+  
+  # Data.table of weights 
+  weights_dt <- fread(file.path(here::here(), "data", "int", "weights", weights_file))
+  # Convert 0-360 to match climate raster
```

Comment:
@saraorofino why do we switch between -180 to 180 and then to 0 to 360 in different scripts?

Reply:
I fixed this so that I adjust the longitude of the polygons to match era5 in the first step, also added a step to adjust the longitude of the secondary weights so everything is in 0-360 when it's saved and it doesn't need to be changed again in step 2